wordpress-mobile/WordPress-Android
# Add screen navigation for Design System

Part of issue https://github.com/wordpress-mobile/WordPress-Android/issues/19802

This PR adds in screen navigation for the `Design System` using the Compose Navigation dependency. It also adds a top navigation bar to all Design System screens. Theming and individual attribute screens (such as Colors, Fonts, DSButton, etc.) will be added in future PRs. 

-----

## To Test:
1. Navigate to the `Me` tab. If you're on a debug build, you should see the `Design System` row. 
  <img width="200" alt="Screenshot 2024-01-18 at 6 42 05 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/86ef5c61-a040-4613-80d8-3aebc76baa5a">

2. Click on the `Design System` row and you should see the following screen:
<img width="202" alt="Screenshot 2024-01-24 at 1 47 45 PM 1" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/1e849a4c-72ca-4ab1-862a-bd27e4d5f058">

3. Click on `Foundation` and it should take you to the screen below. Note: The buttons on this screen will be implemented in future PRs.
<img width="200" alt="Screenshot 2024-01-24 at 1 49 50 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/5444fdbd-2018-4c7a-9bc2-69073adbd240">

4. Click the back button and click on `Components`. You should see the screen below:
<img width="200" alt="Screenshot 2024-01-24 at 1 50 49 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/3c629014-2dd5-4a04-9b07-76765a233049">





-----

## Regression Notes

1. Potential unintended areas of impact

    - None

5. What I did to test those areas of impact (or what existing automated tests I relied on)

    - None

6. What automated tests I added (or what prevented me from doing so)

    - None

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
